R/wwinference.R

```diff
@@ -71,11 +73,20 @@ wwinference <- function(ww_data,
                         mcmc_options = wwinference::get_mcmc_options(),
                         generate_initial_values = TRUE,
                         compiled_model = wwinference::compile_model(),
-                        dist_matrix = NULL) {
+                        dist_matrix = NULL,
+                        bool_spatial_comp = FALSE) {
   # Check that data is compatible with specifications
   check_date(ww_data, model_spec$forecast_date)
   check_date(count_data, model_spec$forecast_date)
 
+  if (bool_spatial_comp == TRUE && is.null(dist_matrix)) {
+    stop(
+      "Spatial Components Desired, but Distance Matrix Not Supplied!!!\n
+          *distance matrix required for current implementation*"
```

Comment:
Do we want this to be this kind of switch? Or do we want it to be a more general switch of whether we infer any sort of a correlation matrix?

Reply:
Makes sense and think this error message should be here for now since we don't have the feature implemented currently.